core/io/resource_uid.cpp

```diff
@@ -99,10 +99,31 @@ ResourceUID::ID ResourceUID::create_id() {
 	}
 }
 
+ResourceUID::ID ResourceUID::create_id_for_path(const String &p_path) {
+	ID id = INVALID_ID;
+	RandomPCG rng;
+
+	const String project_name = GLOBAL_GET("application/config/name");
+	rng.seed(project_name.hash64() * p_path.hash64());
```

Comment:
Are there possible clashes between different combinations of project name and path that could lead to the same seed?

Reply:
Well, the hashes are not guaranteed to be unique, but since they are multiplied, they'd both need to both have a collision.

The ID is guaranteed to be unique within a project though, so this will very rarely be a problem. The only case where something can break is when 2 projects somehow generate the same ID for different resources and then one is copied to the other project.